apps/ensrainbow/Dockerfile

```diff
@@ -5,56 +5,91 @@ RUN apt-get update && \
     npm install -g pnpm
 WORKDIR /app
 
-# Data stage
-FROM base AS ensrainbow.data
-RUN wget -nv https://bucket.ensrainbow.io/ens_names.sql.gz
-RUN wget -nv https://bucket.ensrainbow.io/ens_names.sql.gz.sha256sum
+###################
+## Data Stage — heavily cached remote data
+###################
+
+FROM base AS db-data
+ADD "https://bucket.ensrainbow.io/ens_names.sql.gz" ./ens_names.sql.gz
+ADD "https://bucket.ensrainbow.io/ens_names.sql.gz.sha256sum" ./ens_names.sql.gz.sha256sum
 RUN sha256sum -c ens_names.sql.gz.sha256sum
 
-# Runtime dependencies stage
-FROM base AS ensrainbow.runtime_deps
-# Copy pnpm workspace configuration files from the root directory
+###################
+## Ingest Dependencies Stage — minimal deps for ingest script
+###################
+
+FROM base AS ingest-deps
+
+# copy pnpm files from the root directory
+COPY package.json pnpm-lock.yaml pnpm-workspace.yaml ./
+
+# copy just the dependent packages (sync with package.json)
+# TODO: this could be even more minimal
+COPY packages/ensrainbow-sdk ./packages/ensrainbow-sdk
+COPY packages/ensnode-utils ./packages/ensnode-utils
+
+# cwd to ensrainbow
+WORKDIR /app/apps/ensrainbow
+# copy ensrainbow package.json
+COPY apps/ensrainbow/package*.json ./
+
+# need cli
+COPY apps/ensrainbow/src/cli.ts ./src/
+
+# TODO: have to copy all commands because cli.ts imports all of them
+COPY apps/ensrainbow/src/commands ./src/commands/
+
+# TODO: cli imports lib/server, so have to include that too
+COPY apps/ensrainbow/src/lib ./src/lib/
+
+# need all utils
+COPY apps/ensrainbow/src/utils ./src/utils/
```

Comment:
So it copies whole `src` in the end, right?

Reply:
in the end, yes. goal would be to minimize the dependencies further by restructuring the ingest script